### PR TITLE
fix(settlements)!: SettlementModel currency is mandatory

### PR DESCRIPTION
**Pull request by vgenev**
fix: settlementModel has currency as required
BREAKING CHANGE: currency is mandatory on /post settlementModels
creation of settlementModel on startup has been removed

### CODE REVIEW

File: src/api/ledgerAccountTypes/handler.js [hunk 1]
```diff
@@ -25,32 +25,32 @@
...
 const LedgerAccountTypesService = require('../../domain/ledgerAccountTypes')
-// const Sidecar = require('../../lib/sidecar')
-// const ErrorHandler = require('@mojaloop/central-services-error-handling')
+const Sidecar = require('../../lib/sidecar')
+const ErrorHandler = require('@mojaloop/central-services-error-handling')
 
 const getAll = async function () {
   return LedgerAccountTypesService.getAll()
 }
 // TODO: Below code is commented since there is no requirement to create ledgerAccountType using API.
```
Comment (mdebarros) [inline src/api/ledgerAccountTypes/handler.js:34]: Shouldn't this comment be removed?

File: src/api/settlementModels/routes.js [hunk 2]
```diff
@@ -72,9 +71,9 @@ module.exports = [
           settlementGranularity: Joi.string().required().valid(...settlementGranularityList).description('Granularity type for the settlement model GROSS or NET'),
           settlementInterchange: Joi.string().required().valid(...settlementInterchangeList).description('Interchange type for the settlement model BILATERAL or MULTILATERAL'),
           settlementDelay: Joi.string().required().valid(...settlementDelayList).description('Delay type for the settlement model IMMEDIATE or DEFERRED'),
-          currency: Joi.string().valid(...currencyList).description('Currency code'),
+          currency: Joi.string().required().valid(...currencyList).description('Currency code'),
           requireLiquidityCheck: Joi.boolean().required().description('Liquidity Check boolean'),
-          ledgerAccountType: Joi.string().required().valid(...ledgerAccountList).description('Account type for the settlement model POSITION, SETTLEMENT or INTERCHANGE_FEE'),
+          ledgerAccountType: Joi.string().required().description('Account type for the settlement model POSITION, SETTLEMENT or INTERCHANGE_FEE'),
```
Comment (mdebarros) [inline src/api/settlementModels/routes.js:76]: The description needs to be updated.

`Account type for the settlement model POSITION, SETTLEMENT or INTERCHANGE_FEE` -> `Account type for the settlement model`

File: src/domain/settlement/index.js [hunk 3]
```diff
@@ -28,38 +28,27 @@
...
-      for (const item of existingSettlementModels) {
-        if (item.name === settlementModel.name) {
-          throw new Error(`Settlement model: '${settlementModel.name}' already exists`)
-        }
-        if ((item.name === 'CGS' && settlementModel.name === 'DEFERREDNET') || (item.name === 'DEFERREDNET' && settlementModel.name === 'CGS')) {
-          throw new Error(`Settlement model: '${settlementModel.name}' can't be used with the existing settlement model '${item.name}'`)
-        }
-      }
-    }
+    await ParticipantService.validateHubAccounts(settlementModel.currency)
```
Comment (mdebarros) [inline src/domain/settlement/index.js:37]: Please add a comment indicating why this is put here.

File: test/integration/helpers/settlementModels.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  await Cache.initCache()
+
+  try {
+    await Model.createSettlementModel(settlementModels[0])
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+// exports.deletePreparedData = async (participantName) => {
```
Comment (mdebarros) [inline test/integration/helpers/settlementModels.js:73]: Do we need this?

If so, please add a comment as to why its here.

**Review by mdebarros — CHANGES_REQUESTED** [review]
It looks good! Some minor comments for you to address.

Also, please add an additional comment to the POST /settlementModels API that it will create any associated ledgerAccountTypes for every participant that matches the settlementModel's currency

File: src/domain/settlement/index.js [hunk 5]
```diff
@@ -28,38 +28,28 @@
...
-      for (const item of existingSettlementModels) {
-        if (item.name === settlementModel.name) {
-          throw new Error(`Settlement model: '${settlementModel.name}' already exists`)
-        }
-        if ((item.name === 'CGS' && settlementModel.name === 'DEFERREDNET') || (item.name === 'DEFERREDNET' && settlementModel.name === 'CGS')) {
-          throw new Error(`Settlement model: '${settlementModel.name}' can't be used with the existing settlement model '${item.name}'`)
-        }
-      }
-    }
+    // check for existing hub account with the settlementModel currency
```
Comment (mdebarros) [inline src/domain/settlement/index.js:37]: But why? :D

**Review by mdebarros — CHANGES_REQUESTED** [review]
See minor comment.

File: src/domain/settlement/index.js (hunk 5 above)
Comment (vgenev) [reply src/domain/settlement/index.js:37]: :D

**Review by mdebarros — APPROVED** [review]

**Review by mdebarros — APPROVED** [review]
+1

**Comment by lewisdaly** [comment]
Nice title!

**Pull request merged**